### DISCUSSION TITLE
sonoff rf bridge: Getting a different raw data on every button press

### DISCUSSION THREAD

**Discussion by danielo515** [arendst/Tasmota#13556] [Q&A] [answered]
Hello.
I just successfully flashed tasmota to my rfbridge r2 (v2). I didn't flashed the portish firmware because I am not sure if I am going to need it.
I am trying to get the codes of my ring bell, but every time I push the button I get a different code. I suppose it is using rolling codes, and I am not sure if I can use them in any way.

Worst case scenario, since this is the only RF device on my house I can just react to any RF signal, but It may cause confusion if anyone ever activates any other RF.

The codes I am getting look like this: 

{"Time":"2021-11-03T18:01:44","RfRaw":{"Data":"AAA42E7601AA056400001055"}}
{"Time":"2021-11-03T18:01:50","RfRaw":{"Data":"AAA42ED801BA0528C4BE6055"}}
{"Time":"2021-11-03T18:01:58","RfRaw":{"Data":"AAA42CA201A00570401E3055"}}
{"Time":"2021-11-03T18:02:10","RfRaw":{"Data":"AAA42E7601A20570000C3055"}}

**Comment by ascillato** [comment]
Hi,

If you use the normal way with RfRaw disabled, are you able to see anything in the console? RfRaw contains several things besides the button. For example, the width of the pulses, etc. If you use the normal mode, you should see a JSON with the code sent.

FYI: Portish firmware behaves the same as stock but it is more efficient (less undetected codes) plus it recognizes more protocols. It is safe and recommended for performance to update.

**Reply by danielo515** [reply]
I didn't tested with rf disabled, how can I disable it?

I did not installed the polish firmware yet because I didn't found clear instructions for rfbridge r2 V2.2, but if you link them I'll try it for sure

**Reply by ascillato** [reply]
On the console type `rfraw 0`, then type `RfKey1 2` to learn the code into the key1. After issuing rfkey command, press the button so as to receive the code and learn it.

**Reply by danielo515** [reply]
Thanks for the instructions @ascillato , here is what I got:

```
20:30:41.247 CMD: rfraw 0
20:30:41.254 MQT: stat/sonoff_bridge/RESULT = {"RfRaw":"OFF"}
20:30:49.364 CMD: rfkey1 2
20:30:49.372 MQT: stat/sonoff_bridge/RESULT = {"RfKey1":"Start learning"}
20:31:05.719 MQT: stat/sonoff_bridge/RESULT = {"RfKey1":"Learned"}
20:31:05.825 MQT: tele/sonoff_bridge/RESULT = {"Time":"2021-11-03T20:31:05","RfReceived":{"Sync":11914,"Low":1588,"High":318,"Data":"000000","RfKey":"None"}}
```

Seems that it is not able to learn it?

**Reply by danielo515** [reply]
So, after learning it and clicking the button several times I get something on the result, but it is still different on each entry:

```
20:31:05.825 MQT: tele/sonoff_bridge/RESULT = {"Time":"2021-11-03T20:31:05","RfReceived":{"Sync":11914,"Low":1588,"High":318,"Data":"000000","RfKey":"None"}}
20:32:30.289 MQT: tele/sonoff_bridge/RESULT = {"Time":"2021-11-03T20:32:30","RfReceived":{"Sync":12002,"Low":466,"High":1300,"Data":"C42010","RfKey":"None"}}
20:32:32.394 MQT: tele/sonoff_bridge/STATE = {"Time":"2021-11-03T20:32:32","Uptime":"0T02:50:09","UptimeSec":10209,"Heap":26,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":19,"MqttCount":1,"Wifi":{"AP":1,"SSId":"IOT-UP","BSSId":"18:D6:C7:E8:CA:3C","Channel":11,"Mode":"11n","RSSI":100,"Signal":-35,"LinkCount":1,"Downtime":"0T00:00:03"}}
20:32:48.161 MQT: tele/sonoff_bridge/RESULT = {"Time":"2021-11-03T20:32:48","RfReceived":{"Sync":11372,"Low":408,"High":1378,"Data":"C01000","RfKey":"None"}}
20:33:06.405 MQT: tele/sonoff_bridge/RESULT = {"Time":"2021-11-03T20:33:06","RfReceived":{"Sync":11420,"Low":468,"High":1300,"Data":"C40000","RfKey":"None"}}
20:33:11.589 MQT: tele/sonoff_bridge/RESULT = {"Time":"2021-11-03T20:33:11","RfReceived":{"Sync":11428,"Low":440,"High":1334,"Data":"C00820","RfKey":"None"}}
20:33:18.441 MQT: tele/sonoff_bridge/RESULT = {"Time":"2021-11-03T20:33:18","RfReceived":{"Sync":11372,"Low":426,"High":1360,"Data":"400000","RfKey":"None"}}
20:33:25.701 MQT: tele/sonoff_bridge/RESULT = {"Time":"2021-11-03T20:33:25","RfReceived":{"Sync":11354,"Low":424,"High":1374,"Data":"008620","RfKey":"None"}}
```

**Reply by danielo515** [reply]
This is interesting...
If I go to the main menu and click on the key 1 it sends the learnt RFKey and the doorbell rings, so at least the captured code still works. I guess that means it does not have rolling codes no? Because otherwise that code will be no longer valid.

However my use-case it is not to make my bell to ring, but to capture when anyone presses the button

**Reply by ascillato** [reply]
Yes, your Sonoff Bridge, that is r2 (v2), is not flashable with Portish Firmware yet (https://github.com/arendst/Tasmota/discussions/13283)

When you press the Key1, which code it sends?

From your logs, seems like the frequency of the transmission of the button is not so stable.

**Comment by ascillato** [comment, marked as answer]
If you want, you can check the thread at https://github.com/arendst/Tasmota/discussions/13283 for the discussion in the development of the support to bypass the RF chip. As today is not supported in Tasmota but it is supported in other FW.
